### PR TITLE
Update docs

### PR DESCRIPTION
**Pull request by tmigot**
We can now use the `meta` to get the names of the problems for PureJuMP and ADNLPProblems.

I fixed some more typos when updating the benchmark.

### CODE REVIEW

File: docs/src/benchmark.md [hunk 1]
```diff
@@ -67,3 +67,10 @@ gr()
 
 profile_solvers(stats, costs, costnames)
 ```
+
+Note that it is also possible to select the problems when initializing the problem list by searching into the `OptimizationProblems.meta`.
```
Comment (dpo) [inline docs/src/benchmark.md:71]:
```suggestion
It is also possible to select problems when initializing the problem list by filtering `OptimizationProblems.meta`.
```

File: docs/src/tutorial.md [hunk 2]
```diff
@@ -5,17 +5,22 @@ This package is subdivided in two submodules: `PureJuMP` for the JuMP problems,
 
 ## Problems in JuMP syntax: PureJuMP
 
-You can obtain the list of problems currently defined with `names(OptimizationProblems.PureJuMP)` that returns the list of exported names of the submodule. The symbol `:PureJuMP` is itself in this list.
+You can obtain the list of problems currently defined with `OptimizationProblems.meta[!, :name]` that returns the list of exported names of the submodule.
```
Comment (dpo) [inline docs/src/tutorial.md:8]:
```suggestion
You can obtain the list of problems currently defined with `OptimizationProblems.meta[!, :name]`.
```

File: docs/src/tutorial.md [hunk 3]
```diff
@@ -5,17 +5,22 @@ This package is subdivided in two submodules: `PureJuMP` for the JuMP problems,
...
-problems = setdiff(names(OptimizationProblems.PureJuMP), [:PureJuMP])
+problems = OptimizationProblems.meta[!, :name]
 length(problems)
 ```
 Then, it suffices to select any of this problem to get the JuMP model.
 ``` @example ex1
 jump_model = zangwil3()
 ```
-Note that some of these problems are scalable, i.e., their size depends on some parameters that can be modified.
+Note that some of these problems are scalable, i.e., their size depends on some parameters that can be modified. The list is available once again using the `meta`.
```
Comment (dpo) [inline docs/src/tutorial.md:18]:
```suggestion
Note that certain problems are scalable, i.e., their size depends on parameters that can be modified. The list of those problems is available once again using `meta`.
```

File: docs/src/tutorial.md [hunk 4]
```diff
@@ -31,10 +36,10 @@ This package also offers [ADNLPModel](https://github.com/JuliaSmoothOptimizers/A
 ``` @example ex2
 using ADNLPModels
 ```
-You can obtain the list of problems currently defined with `names(OptimizationProblems.ADNLPProblems)` that returns the list of exported names of the submodule. The symbol `:ADNLPProblems` is itself in this list.
+You can obtain the list of problems currently defined with `OptimizationProblems.meta[!, :name]` that returns the list of exported names of the submodule.
```
Comment (dpo) [inline docs/src/tutorial.md:39]:
```suggestion
You can obtain the list of problems currently defined with `OptimizationProblems.meta[!, :name]`.
```

**Review by dpo — CHANGES_REQUESTED** [review]

**Comment by tmigot** [comment]
Thanks @dpo  for the comments. I merged them directly.

File: docs/src/benchmark.md [hunk 5]
```diff
@@ -67,3 +67,10 @@ gr()
 
 profile_solvers(stats, costs, costnames)
 ```
+
+It is also possible to select problems when initializing the problem list by filtering `OptimizationProblems.meta`.
```
Comment (dpo) [inline docs/src/benchmark.md:71]:
```suggestion
It is also possible to select problems when initializing the problem list by filtering `OptimizationProblems.meta`:
```

File: docs/src/tutorial.md [hunk 6]
```diff
@@ -5,17 +5,22 @@ This package is subdivided in two submodules: `PureJuMP` for the JuMP problems,
...
-problems = setdiff(names(OptimizationProblems.PureJuMP), [:PureJuMP])
+problems = OptimizationProblems.meta[!, :name]
 length(problems)
 ```
 Then, it suffices to select any of this problem to get the JuMP model.
 ``` @example ex1
 jump_model = zangwil3()
 ```
-Note that some of these problems are scalable, i.e., their size depends on some parameters that can be modified.
+Note that certain problems are scalable, i.e., their size depends on parameters that can be modified. The list of those problems is available once again using `meta`.
```
Comment (dpo) [inline docs/src/tutorial.md:18]:
```suggestion
Note that certain problems are scalable, i.e., their size depends on parameters that can be modified. The list of those problems is available once again using `meta`:
```

File: docs/src/tutorial.md [hunk 7]
```diff
@@ -31,10 +36,10 @@ This package also offers [ADNLPModel](https://github.com/JuliaSmoothOptimizers/A
 ``` @example ex2
 using ADNLPModels
 ```
-You can obtain the list of problems currently defined with `names(OptimizationProblems.ADNLPProblems)` that returns the list of exported names of the submodule. The symbol `:ADNLPProblems` is itself in this list.
+You can obtain the list of problems currently defined with `OptimizationProblems.meta[!, :name]`.
```
Comment (dpo) [inline docs/src/tutorial.md:39]:
```suggestion
You can obtain the list of problems currently defined with `OptimizationProblems.meta[!, :name]`:
```

**Review by dpo — COMMENTED** [review]
Looks great, thank you! Just a couple of very minor things. Go ahead and merge after those.

**Pull request merged**